tests/routesTests/PostLeagueTests.py

```diff
@@ -0,0 +1,320 @@
+      
+      self.assertEquals(response.status_code, 400)
+      self.assertTrue('MissingField' in responseData[0])
+      self.assertEquals(responseData[0]['MissingField'], 'name is a required field')
+
+   
+   def test_post_league_missingStart(self):
+      body = {
+         "name": "myLeague",
+         "end": "08-30-2019",
+         "startPos": 3000
+      }
+      response = requests.post(url=self.url, data=json.dumps(body), headers=self.headers)
+      responseData = self.getJson(response)
+
+      self.assertEquals(response.status_code, 400)
+      self.assertTrue('MissingField' in responseData[0])
+      self.assertEquals(responseData[0]['MissingField'], 'start is a required field')
+
+
+   def test_post_league_missingEnd(self):
+      body = {
+         "name": "myLeague",
+         "start": "08-30-2019",
+         "startPos": 3000
+      }
+      response = requests.post(url=self.url, data=json.dumps(body), headers=self.headers)
+      responseData = self.getJson(response)
+
+      self.assertEquals(response.status_code, 400)
+      self.assertTrue('MissingField' in responseData[0])
+      self.assertEquals(responseData[0]['MissingField'], 'end is a required field')
+
+   
+   def test_post_league_invalidName(self):
+      body = {
+         "name": 23113,
+         "start": "08-24-2019",
+         "end": "08-30-2019",
+         "startPos": 3000
+      }
+      response = requests.post(url=self.url, data=json.dumps(body), headers=self.headers)
+      responseData = self.getJson(response)
+
+      self.assertEquals(response.status_code, 400)
+      self.assertTrue('InvalidField' in responseData[0])
+      self.assertEquals(responseData[0]['InvalidField'], 'name is not a string or formatted incorrectly')
+   
+
+   def test_post_league_emptyName(self):
+      body = {
+         "name": '',
+         "start": "08-24-2019",
+         "end": "08-30-2019",
+         "startPos": 3000
+      }
+      response = requests.post(url=self.url, data=json.dumps(body), headers=self.headers)
+      responseData = self.getJson(response)
+
+      self.assertEquals(response.status_code, 400)
+      self.assertTrue('InvalidField' in responseData[0])
+      self.assertEquals(responseData[0]['InvalidField'], 'name must not be empty')
+
+
+   def test_post_league_invalidStart(self):
+      body = {
+         "name": 'myLeague',
+         "start": "08/24/2019",
+         "end": "08-30-2019",
+         "startPos": 3000
+      }
+      response = requests.post(url=self.url, data=json.dumps(body), headers=self.headers)
+      responseData = self.getJson(response)
+
+      self.assertEquals(response.status_code, 400)
+      self.assertTrue('InvalidField' in responseData[0])
+      self.assertEquals(responseData[0]['InvalidField'], 'start date must be in MM-DD-YYYY format')
+
+
+   def test_post_league_invalidEnd(self):
+      body = {
+         "name": 'myLeague',
+         "start": "08-24-2019",
+         "end": "08/30/2019",
```

Comment:
Not really necessary but would this still work if someone put an invalid date like 08-35-2019?

Reply:
Good test idea! Lemme add a test for it and fix it if it breaks.